lib/raygun.messageBuilder.ts

```diff
@@ -31,18 +31,37 @@ type UserMessageData = RawUserData | string | undefined;
 const humanString = require("object-to-human-string");
 const packageDetails = require("../package.json");
 
-function filterKeys(obj: object, filters: string[]): object {
+/**
+ * Filter properties in obj according to provided filters.
+ * Also removes any recursive self-referencing object.
+ * @param obj object to apply filter
+ * @param filters list of keys to filter
+ * @param explored Set that contains already explored nodes, used internally
+ */
+function filterKeys(
```

Comment:
Not sure if it's been checked but it may be useful to check around the repo to see if we've done infinite recursion in the past and fix that up?

Reply:
I couldn't see any other instances in this repo. Do you mean in all the codebases from Raygun?